### PR TITLE
Feature/trailing slash

### LINKED ISSUE #2859: vic-machine: doesn't handle datacenter `/` and doesn't suggest options with `ls`
Running against a local ESX I see the following when target has a trailing `/`, e.g. `https://user:password@esx/`

```
INFO[2016-10-23T16:16:44Z] ### Installing VCH ####
time=2016-10-23T16:16:44.152291463Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/install/validate.CreateNoDCCheck:80]
DEBU[2016-10-23T16:16:44Z] Using to login by username/password
DEBU[2016-10-23T16:16:44Z] Session Environment Info:                     API Type=HostAgent API Version=6.0 Build=3380124 Name=VMware ESXi OS Type=vmnix-x86 Product ID=embeddedEsx UUID= Vendor=VMware, Inc. Version=6.0.0
DEBU[2016-10-23T16:16:44Z] vSphere resource cache populating...
DEBU[2016-10-23T16:16:44Z] Error count populating vSphere cache: (5)
time=2016-10-23T16:16:44.227323863Z level=debug msg=[ END ] [github.com/vmware/vic/lib/install/validate.CreateNoDCCheck:80] [75.013344ms]
ERRO[2016-10-23T16:16:44Z] Datacenter must be specified in --target (e.g. https://addr/datacenter)
time=2016-10-23T16:16:44.227859362Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/install/validate.(*Validator).suggestDatacenter:170]
INFO[2016-10-23T16:16:44Z] Suggesting valid values for datacenter in --target
INFO[2016-10-23T16:16:44Z] Suggested datacenters:
INFO[2016-10-23T16:16:44Z]   "ha-datacenter"
time=2016-10-23T16:16:44.234730178Z level=debug msg=[ END ] [github.com/vmware/vic/lib/install/validate.(*Validator).suggestDatacenter:170] [6.746422ms]
ERRO[2016-10-23T16:16:44Z] Create cannot continue: failed to create validator
ERRO[2016-10-23T16:16:44Z] --------------------
ERRO[2016-10-23T16:16:44Z] vic-machine-linux failed: Datacenter must be specified in --target (e.g. https://addr/datacenter)
```

Also for `ls` with that same target there's no suggestion at all:

```
INFO[2016-10-23T16:18:41Z] ### Listing VCHs ####
INFO[2016-10-23T16:18:41Z] Validating target
ERRO[2016-10-23T16:18:41Z] --------------------
ERRO[2016-10-23T16:18:41Z] Failed to get host system: please specify a datacenter
ERRO[2016-10-23T16:18:41Z] List cannot continue - target validation failed: validation of configuration failed
ERRO[2016-10-23T16:18:41Z] --------------------
ERRO[2016-10-23T16:18:41Z] vic-machine-linux failed: validation of configuration failed
```

Datacenter selection and validation should be a common path for all vic-machine commands as it's an inherent part of the target, which is required for all commands.

### PR DESCRIPTION
**Pull request by mhagen-vmware**

### CODE REVIEW

**Comment by sgairo** [comment]
LGTM

File: tests/test-cases/Group6-VIC-Machine/6-10-List.robot [hunk 1]
```diff
@@ -54,4 +54,8 @@ List with trailing slash
     Should Contain  ${ret}  ID
     Should Contain  ${ret}  PATH
     Should Contain  ${ret}  NAME
-    Should Not Contain  ${ret}  Error
\ No newline at end of file
+    Should Not Contain  ${ret}  Error
+    
+List suggest compute resource
+    ${ret}=  Run  bin/vic-machine-linux ls --target %{TEST_URL} --thumbprint=%{TEST_THUMBPRINT} --user %{TEST_USERNAME} --password=%{TEST_PASSWORD} --compute-resource fakeComputeResource
+    Should Contain  ${ret}  Suggested values for --compute-resource:
```
Comment (emlin) [inline tests/test-cases/Group6-VIC-Machine/6-10-List.robot]: it is nice. Please also add another case to test list function happy path, I mean with correct dc, and no compute resource specified

**Comment by emlin** [comment]
this change lgtm. 👍 
but as George mentioned the suggestion issue for vic-machine ls, please leave that issue open. I'll take care of the rest of it.

**Comment by mhagen-vmware** [comment]
.

**Comment by mhagen-vmware** [comment]
lgtm

**Pull request merged**